tests/unit/server/bundle_manager/make_bundles_test.py

```diff
@@ -0,0 +1,272 @@
+
+        threads = self.bundle_manager._make_bundles()
+        self.assertTrue(self.bundle_manager._is_making_bundles())
+        for t in threads:
+            t.join()
+        self.assertFalse(self.bundle_manager._is_making_bundles())
+
+        bundle = self.bundle_manager._model.get_bundle(bundle.uuid)
+        self.assertEqual(bundle.state, State.READY)
+
+    def test_single_dependency(self):
+        parent = RunBundle.construct(
+            targets=[],
+            command='',
+            metadata=BASE_METADATA,
+            owner_id=self.user_id,
+            uuid=generate_uuid(),
+            state=State.READY,
+        )
+        with open(self.codalab_manager.bundle_store().get_bundle_location(parent.uuid), "w+") as f:
+            f.write("hello world")
+        bundle = MakeBundle.construct(
+            targets=[],
+            command='',
+            metadata=BASE_METADATA_MAKE_BUNDLE,
+            owner_id=self.user_id,
+            uuid=generate_uuid(),
+            state=State.STAGED,
+        )
+        bundle.dependencies = [
+            Dependency(
+                {
+                    "parent_uuid": parent.uuid,
+                    "parent_path": "",
+                    "child_uuid": bundle.uuid,
+                    "child_path": "src",
+                }
+            )
+        ]
+        self.bundle_manager._model.save_bundle(parent)
+        self.bundle_manager._model.save_bundle(bundle)
+
+        threads = self.bundle_manager._make_bundles()
+        self.assertTrue(self.bundle_manager._is_making_bundles())
+        for t in threads:
+            t.join()
+        self.assertFalse(self.bundle_manager._is_making_bundles())
+
+        bundle = self.bundle_manager._model.get_bundle(bundle.uuid)
+
+        self.assertEqual(bundle.state, State.READY)
+
+        with open(
+            os.path.join(
+                self.codalab_manager.bundle_store().get_bundle_location(bundle.uuid), "src"
+            ),
+            "r",
+        ) as f:
+            self.assertEqual(f.read(), "hello world")
+
+    def test_multiple_dependencies(self):
+        parent1 = RunBundle.construct(
+            targets=[],
+            command='',
+            metadata=BASE_METADATA,
+            owner_id=self.user_id,
+            uuid=generate_uuid(),
+            state=State.READY,
+        )
+        self.bundle_manager._model.save_bundle(parent1)
+        with open(self.codalab_manager.bundle_store().get_bundle_location(parent1.uuid), "w+") as f:
+            f.write("hello world 1")
+        parent2 = RunBundle.construct(
+            targets=[],
+            command='',
+            metadata=BASE_METADATA,
+            owner_id=self.user_id,
+            uuid=generate_uuid(),
+            state=State.READY,
+        )
+        self.bundle_manager._model.save_bundle(parent2)
+        with open(self.codalab_manager.bundle_store().get_bundle_location(parent2.uuid), "w+") as f:
```

Comment:
avoid code duplication here; can create a general helper function for creating K dependencies